cabal-install/Distribution/Client/Outdated.hs

```diff
@@ -197,14 +198,15 @@ listOutdated deps pkgIndex (ListOutdatedSettings ignorePred minorPred) =
     lookupLatest :: Dependency -> [Version]
     lookupLatest dep@(Dependency pname vr _)
       | minorPred (depPkgName dep) =
-        map packageVersion $ lookupDependency pkgIndex  pname (relaxMinor vr)
+        map packageVersion $ lookupDependency pkgIndex  pname =<< maybeToList (relaxMinor vr)
       | otherwise                  =
         map packageVersion $ lookupPackageName pkgIndex (depPkgName dep)
 
-    relaxMinor :: VersionRange -> VersionRange
+    relaxMinor :: VersionRange -> Maybe VersionRange
```

Comment:
I think this i better to still return only `VersionRange`. I.e. if there isn't `last`, just return `vr` as is.